packages/react-components/react-utilities/src/utils/isHTMLElement.ts

```diff
@@ -7,11 +7,76 @@
  * might be problematic while operating with [multiple realms](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/instanceof#instanceof_and_multiple_realms)
  *
  */
-export function isHTMLElement(element?: unknown): element is HTMLElement {
+export function isHTMLElement<ConstructorName extends HTMLElementConstructorName = 'HTMLElement'>(
+  element?: unknown,
+  options?: { constructorName?: ConstructorName },
+): element is InstanceType<(typeof globalThis)[ConstructorName]> {
   const typedElement = element as Node | null | undefined;
   return Boolean(
     typedElement !== null &&
```

Comment:
Since you're using optional chaining on the next line I don't think you need this check.